Examples/ResourceLeak/README.md

```diff
@@ -0,0 +1,46 @@
+## Procedure-local Bugs of Resource Leaks
+
+**CSharpCodeAnalyzer** is able to detect resource leaks which include the C# scenarios enumerated below. For further detail on its capabilities, see [here](http://www.eecs.qmul.ac.uk/~ddino/papers/nasa-infer.pdf). 
+
+## Supported scenarios
+
+### 1. Deallocation of IDisposable Local Variables: 
+
+Instances of classes deriving from IDisposable (such as input streams, output streams, http connections, and cursors) create resource leaks if they are not closed or disposed of.
+
+```c#
+public void ResourceLeakBad(){
+    StreamReader sr = new StreamReader("whatever.txt");  
+    string data = sr.ReadToEnd();
+    Console.WriteLine(data);
+    // FIXME: should close the stream by calling sr.Close().
+}
+```
+							
+### 2. Resource Allocation in Method Invocation: 
+Resource leaks may occur when a resource is allocated as an argument of a method invocation (including methods which allocate resources such as constructors); should an exception occur during the method invocation, the resource will leak.	
+
+For example, in the following example:
```

Comment:
say "For example, if the GZipStream constructor invocation below throws an exception, the allocated FileStream resource will leak."

Reply:
(and delete the sentence below the example)

Reply:
Resolved in the update